apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx

```diff
@@ -1,9 +1,132 @@
+import React from 'react';
+import { defineMessage, FormattedMessage, IntlShape, MessageDescriptor, useIntl } from 'react-intl';
+import CommonTransactionStepItem from '@common/components/transaction-steps/common-transaction-step';
 import useActiveWallet from '@hooks/useActiveWallet';
 import useContractService from '@hooks/useContractService';
-import { DisplayStrategy, EarnPermission, WithdrawType } from 'common-types';
-import React from 'react';
-import { FormattedMessage } from 'react-intl';
-import { Modal } from 'ui-library';
+import { AmountsOfToken, DisplayStrategy, EarnPermission, EarnPosition, Token, WithdrawType } from 'common-types';
+import { ContainerBox, Modal, Wallet2Icon, UnlockIcon, Typography, colors, Skeleton } from 'ui-library';
+import TokenIcon from '@common/components/token-icon';
+import { formatCurrencyAmount, formatUsdAmount } from '@common/utils/currency';
+import { useEarnManagementState } from '@state/earn-management/hooks';
+import { isUndefined } from 'lodash';
+import useEstimateMarketWithdraw from '../hooks/useEstimateMarketWithdraw';
+
+const WithdrawAmountItem = ({
+  title,
+  amount,
+  token,
+  intl,
+  isLoading,
+  isNegative,
+}: {
+  title: MessageDescriptor;
+  amount?: AmountsOfToken;
+  token?: Token;
+  intl: IntlShape;
+  isLoading?: boolean;
+  isNegative?: boolean;
+}) => (
+  <ContainerBox flexDirection="column" gap={1}>
+    <Typography variant="bodySmallRegular">{intl.formatMessage(title)}</Typography>
+    <ContainerBox gap={2} alignItems="center">
+      <TokenIcon token={token} size={6} />
+      <Typography variant="bodyBold" color={({ palette }) => colors[palette.mode].typography.typo2} lineHeight={1}>
+        {isLoading ? (
```

Comment:
this probably should not have the () around the USD price, its a great use case for TokenAmount :)

Reply:
Just followed the figma here 😅 , but i'll prioritize your PR so we can implement it here